src/python/deepgnn/graph_engine/snark/decoders.py

```diff
@@ -6,31 +6,285 @@
 import json
 import logging
 import csv
-from enum import Enum
-from typing import Any, Dict
+import numpy as np
 
 logger = logging.getLogger()
 
 
-class DecoderType(Enum):
-    """Decoder types supported by converter."""
-
-    JSON = "json"
-    TSV = "tsv"
-
-    def __str__(self):
-        """Convert instance to string."""
-        return self.value
-
-
 class Decoder(abc.ABC):
     """Interface to convert one line of text into node object."""
 
+    convert_map = {
+        "double_feature": np.float64,
+        "float_feature": np.float32,
+        "float64_feature": np.float64,
+        "float32_feature": np.float32,
+        "float16_feature": np.float16,
+        "uint64_feature": np.uint64,
+        "int64_feature": np.int64,
+        "uint32_feature": np.uint32,
+        "int32_feature": np.int32,
+        "uint16_feature": np.uint16,
+        "int16_feature": np.int16,
+        "uint8_feature": np.uint8,
+        "int8_feature": np.int8,
+    }
+
+    def set_metadata(self, metadata: dict):
```

Comment:
This is for the header line - so users can specify that all nodes will have type 0 or weight 1 in the meta file and it will be shared to all partitions. This approach is much better than doing the first line of the graph file since it is more readable - also it is harder to get the first line of the text file shared to all partitions. Since the meta file it is in json format users can also leverage this.